addons/web/static/src/search/filter_menu/custom_filter_item.xml

```diff
@@ -7,91 +7,91 @@
-                                date="condition.value[1]"
-                                onDateTimeChanged="date => this.onDateTimeChanged(condition, 1, date)"
+                            <input t-elif="['integer', 'id'].includes(fieldType)"
+                                class="o_input"
+                                step="1"
+                                type="number"
+                                t-model="condition.displayedValue"
+                                t-on-change="ev => this.onValueChange(condition, ev)"
                             />
-                        </t>
-                        <select t-elif="fieldType === 'selection'" class="o_input"
-                            t-on-change="ev => this.onValueChange(condition, ev)"
-                            >
-                            <option t-foreach="fields[condition.field].selection" t-as="option" t-key="option_index"
-                                t-att-value="option[0]"
-                                t-esc="option[1]"
+                            <input t-else=""
+                                type="text"
+                                class="o_input"
+                                t-model="condition.displayedValue"
+                                t-on-change="ev => this.onValueChange(condition, ev)"
                             />
-                        </select>
-                        <!-- @todo (DAM) I think that the localization should be better consisered below -->
-                        <input t-elif="fieldType === 'float'"
-                            class="o_input"
-                            step="0.01"
-                            t-att-type="DECIMAL_POINT === '.' ? 'number' : 'text'"
-                            t-attf-title="Number using {{ DECIMAL_POINT }} as decimal separator."
-                            t-attf-pattern="[0-9]+([\\{{ DECIMAL_POINT }}][0-9]+)?"
-                            t-model="condition.displayedValue"
-                            t-on-change="ev => this.onValueChange(condition, ev)"
-                        />
-                        <input t-elif="['integer', 'id'].includes(fieldType)"
-                            class="o_input"
-                            step="1"
-                            type="number"
-                            t-model="condition.displayedValue"
-                            t-on-change="ev => this.onValueChange(condition, ev)"
-                        />
-                        <input t-else=""
-                            type="text"
-                            class="o_input"
-                            t-model="condition.displayedValue"
-                            t-on-change="ev => this.onValueChange(condition, ev)"
-                        />
-                    </span>
+                        </span>
+                    </div>
                     <i t-if="conditions.length gt 1"
```

Comment:
I know that it was already there, but I'd prefer to don't bind button's actions to icons.
Could you use a `<button>`tag and place an `<i>`icons inside?